packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js

```diff
@@ -225,54 +225,54 @@ class CloudFormation {
         const self = this;
 
         this.eventStartTime = new Date();
-
         return new Promise((resolve, reject) => {
-          cfnModel.describeStacks(cfnStackCheckParams, err => {
-            if (err) {
-              reject(new Error("Project stack doesn't exist"));
-              context.print.info(err.stack);
-            }
-            const cfnParentStackParams = {
-              StackName: stackName,
-              TemplateURL: templateURL,
-              Capabilities: ['CAPABILITY_NAMED_IAM', 'CAPABILITY_AUTO_EXPAND'],
-              Parameters: [
-                {
-                  ParameterKey: 'DeploymentBucketName',
-                  ParameterValue: deploymentBucketName,
-                },
-                {
-                  ParameterKey: 'AuthRoleName',
-                  ParameterValue: authRoleName,
-                },
-                {
-                  ParameterKey: 'UnauthRoleName',
-                  ParameterValue: unauthRoleName,
-                },
-              ],
-            };
-
-            cfnModel.updateStack(cfnParentStackParams, updateErr => {
-              self.readStackEvents(stackName);
-
-              const cfnCompleteStatus = 'stackUpdateComplete';
-              if (updateErr) {
-                console.error('Error updating cloudformation stack');
-                reject(updateErr);
-              }
-              cfnModel.waitFor(cfnCompleteStatus, cfnStackCheckParams, completeErr => {
-                if (self.pollForEvents) {
-                  clearInterval(self.pollForEvents);
-                }
-                if (completeErr) {
+          this.describeStack(cfnStackCheckParams)
```

Comment:
If we are doing retry calls on describeStack - would it be best to add `retryDelayOptions` on the cloudformation constructor?

Reply:
exposing it in constructor would make sense only when we add retry options to all the CFN calls, not just for describeStack